src/components/MultiColumnMeasures.tsx

```diff
@@ -1,163 +1,395 @@
-      )
-    );
-    updateColumnIsPartOf(columnId, null);
-  };
-
-  const removeCard = (cardId: string) => {
-    const card = termCards.find((c) => c.id === cardId);
-    if (card) {
-      card.mappedColumns.forEach((columnId) => {
-        updateColumnIsPartOf(columnId, null);
+    },
+    [currentVariable]
+  );
+
+  const handleColumnSelect = useCallback(
+    (cardId: string, columnId: string | null) => {
+      if (!columnId || !currentVariable) return;
+
+      setVariableStates((prev) => {
+        const currentCards = prev[currentVariable.identifier]?.termCards || [];
+        const updatedCards = currentCards.map((card) =>
+          card.id === cardId && !card.mappedColumns.includes(columnId)
+            ? { ...card, mappedColumns: [...card.mappedColumns, columnId] }
+            : card
+        );
+
+        return {
+          ...prev,
+          [currentVariable.identifier]: {
+            terms: prev[currentVariable.identifier]?.terms || [],
+            termCards: updatedCards,
+          },
+        };
       });
-    }
 
-    const newCards = termCards.filter((termCard) => termCard.id !== cardId);
-    if (newCards.length === 0) {
-      setTermCards([{ id: generateID(), term: null, mappedColumns: [] }]);
-      handlePaginationChange({} as React.ChangeEvent<unknown>, 1);
-    } else {
-      setTermCards(newCards);
-      const newTotalPages = Math.ceil(newCards.length / itemsPerPage);
-      if (currentPage > newTotalPages) {
-        handlePaginationChange({} as React.ChangeEvent<unknown>, newTotalPages);
+      const card = currentTermCards.find((c) => c.id === cardId);
+      if (card?.term) {
+        updateColumnIsPartOf(columnId, {
+          identifier: card.term.identifier,
+          label: card.term.label,
+        });
       }
-    }
-  };
+    },
+    [currentVariable, currentTermCards, updateColumnIsPartOf]
+  );
+
+  const removeColumnFromCard = useCallback(
+    (cardId: string, columnId: string) => {
+      if (!currentVariable) return;
+
+      setVariableStates((prev) => {
+        const currentCards = prev[currentVariable.identifier]?.termCards || [];
+        const updatedCards = currentCards.map((card) =>
+          card.id === cardId
+            ? {
+                ...card,
+                mappedColumns: card.mappedColumns.filter((id) => id !== columnId),
+              }
+            : card
+        );
+
+        return {
+          ...prev,
+          [currentVariable.identifier]: {
+            terms: prev[currentVariable.identifier]?.terms || [],
+            termCards: updatedCards,
+          },
+        };
+      });
+      updateColumnIsPartOf(columnId, null);
+    },
+    [currentVariable, updateColumnIsPartOf]
+  );
 
-  const columnsAssigned = () => {
-    if (allMappedColumns.length === 0) return 'No columns assigned';
-    if (allMappedColumns.length === 1) return '1 column assigned';
-    return `${allMappedColumns.length} columns assigned`;
-  };
+  const removeCard = useCallback(
```

Comment:
why are some functions called "handle..." and others like this one just "remove..."? These are both event handlers, no?